app/views/report_templates/generate.html.erb

```diff
@@ -11,7 +11,16 @@
                 (_('This will generate a report %s. Based on its definition, it can take a long time to process.') % h(@template.name)) +
                 '</p>').html_safe) %>
 
-  <%= text_f f, :generate_at, label: _('Generate at'), label_help: _('Generate report on a given time. Both date and time are required. Please use format yyyy-mm-dd HH:MM +000') %>
+  <div id="report_generate_at"></div>
+  <%= mount_react_component('DateTime', '#report_generate_at', {
+      label: _('Generate at'),
+      label_help: _('Generate report on a given time.'),
+      id: 'report_generate_at_datetime',
+      value: ( f.object.generate_at || Time.now ).try(:iso8601),
+      hideValue: f.object.generate_at.blank?,
+      inputProps: { name: 'report_template_report[generate_at]' }
```

Comment:
I would like to discuss in here, what other people think about the API for the react form inputs like this.
What we agree upon, I would put to a generic form field component in followup PR.
Thoughts @theforeman/ui-ux ?

Reply:
I did not fully understand your concern, but here a clarification about forms and react in foreman:

In the short term, we'd like to combine rails forms with react inputs, like dateTime, password strength etc. so the react input component is mixed with the rails form itself, and its value will send after a click on submit and be a part of the rails form request. therefore  that is the reason we use the name attribute in these inputs.

In the long term we wish to use react form component in order to replace the entire form behavior.
In order to achieve that we already use `redux-form` (i.e bookmarks), but IMHO it's a bit overkill (forms shouldn't be statefull, and mutating the store due to form changes looks redundant) so we're considering switching to other package, like formik.

Reply:
Is there any discourse thread about it? Integration to the rails form is exactly the point I were trying to make and for that we need a name. But I would rather have the discussion on discourse.

Reply:
Are you suggesting adding a generic input component which integrates with rails forms? 
looks to me a bit overkill, because this integration relay only related to one attribute.

Reply:
It would seems more rails way and easier to use for rails developers, and it is not ment just for this component :)
Follow up> https://community.theforeman.org/t/react-form-inputs-integration-to-rails-forms/13549